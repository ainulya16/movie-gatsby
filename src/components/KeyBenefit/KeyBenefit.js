import React from 'react';
import PropTypes, { any, objectOf } from 'prop-types';
import styled from 'styled-components';
import { Accordion } from '@aia-digital/ui-library';

const KeyBenefitStyled = styled.div`
  .titleBenefit {
    text-align: center;
  }

  .titleBenefit div:nth-child(1) {
    color: #d31145;
  }

  .titleBenefit div:nth-child(2) {
    font-weight: bold;
    margin-bottom: 20px;
    font-size: 25px;
  }
`;

const getKeyBenefits = product => {
  let result = [];
  const benefits = product && product.benefits ? product.benefits : [];
  result = benefits.map(item => ({
    header: item.name.en,
    content: item.description.en,
  }));
  return result;
};

const KeyBenefit = props => {
  const { desc, title, data } = props;
  const benefits = getKeyBenefits(data);
  return (
    <KeyBenefitStyled>
      <div className="keyBenefitContent">
        <div className="titleBenefit">
          <div>{desc}</div>
          <div>{title}</div>
        </div>
      </div>
      <Accordion data={benefits} />
    </KeyBenefitStyled>
  );
};

KeyBenefit.propTypes = {
  desc: PropTypes.string,
  title: PropTypes.string,
  data: objectOf(any),
};

KeyBenefit.defaultProps = {
  desc: '',
  title: '',
  data: {},
};

export default KeyBenefit;
