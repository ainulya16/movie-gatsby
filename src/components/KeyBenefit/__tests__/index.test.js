import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import KeyBenefit from '../KeyBenefit';

describe('<KeyBenefit />', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<KeyBenefit />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have props data benefits', () => {
    const benefits = {
      benefits: [
        {
          name: {
            en: 'name',
          },
          description: {
            en: 'description',
          },
        },
      ],
    };
    const component = mount(<KeyBenefit data={benefits} />);
    const res = component.prop('data');
    expect(res).toBe(benefits);
  });
});
