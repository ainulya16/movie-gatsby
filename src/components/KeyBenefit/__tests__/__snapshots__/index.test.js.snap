// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeyBenefit /> should match the snapshot 1`] = `
.c1 {
  width: 100%;
}

.c0 .titleBenefit {
  text-align: center;
}

.c0 .titleBenefit div:nth-child(1) {
  color: #d31145;
}

.c0 .titleBenefit div:nth-child(2) {
  font-weight: bold;
  margin-bottom: 20px;
  font-size: 25px;
}

<div
  className="c0"
>
  <div
    className="keyBenefitContent"
  >
    <div
      className="titleBenefit"
    >
      <div>
        
      </div>
      <div>
        
      </div>
    </div>
  </div>
  <div
    className="c1"
    style={null}
  />
</div>
`;
